### PR TITLE
Pass python version to docker container, see issue #154

### PR DESCRIPTION
**Pull request by kaelancotter**
Local building of the container indicates TRAVIS_PYTHON_VERSION is properly passed to the conda environment.  If this does work, travis ci might still fail on the sge 2.7 job.  I figure first attempt is free, but if something particularly unusual shows up in the CI logs, I'll try and setup Travis on my fork.  Wish me luck. ;)

### CODE REVIEW

**Comment by kaelancotter** [comment]
I think we can check for the environment variable in the start-sge.sh script and export a default value if not already set.

**Comment by kaelancotter** [comment]
Can/should I change this to WIP for the time being...  I wasn't entirely expecting the current behavior.  There might be some additional 3.x dependency in the testing support scripts.

**Comment by guillaumeeb** [comment]
> I think we can check for the environment variable in the start-sge.sh 

I would prefer to have that directly in the Docker images if possible.

> Can/should I change this to WIP for the time being...

Yes feeling free to do this if you need more time!

**Comment by kaelancotter** [comment]
Looks like part of the issue was the use of  http.server vs SimpleHTTPServer in the run-master and run-slave scripts.  As far as locally running the various testing environments, is it worth building a new wrapper script? (This could/would also be a place for setting TRAVIS_PYTHON_VERSION)  LMK if there's already a unified/single script way for locally running the docker-compose based tests.

**Comment by kaelancotter** [comment]
Side thought, should the slave containers be tied to TRAVIS_PYTHON_VERSION as in above?  Could be limited to just master.

File: ci/sge/Dockerfile-master [hunk 1]
```diff
@@ -9,7 +9,9 @@ RUN curl -o miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-L
     /opt/anaconda/bin/conda clean -tipy && \
     rm -f miniconda.sh
 ENV PATH /opt/anaconda/bin:$PATH
-RUN conda install -c conda-forge dask distributed pytest && conda clean -tipy
+ARG TRAVIS_PYTHON_VERSION
+ENV TRAVIS_PYTHON_VERSION=$TRAVIS_PYTHON_VERSION
```
Comment (guillaumeeb) [inline ci/sge/Dockerfile-master]: Is the ENV line really needed, doesn't ARG suffice?

File: ci/sge/docker-compose.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ services:
     build:
       context: .
       dockerfile: Dockerfile-master
+      args:
+        - TRAVIS_PYTHON_VERSION
```
Comment (guillaumeeb) [inline ci/sge/docker-compose.yml]: So what will happen in local testing if TRAVIS_PYTHON_VERSION is not set?

File: ci/sge/run-master.sh [hunk 3]
```diff
@@ -20,4 +20,8 @@ bash add_worker.sh dask.q slave_two 4
 
 sudo service gridengine-master restart
 
-python -m http.server 8888
+if [ "$TRAVIS_PYTHON_VERSION" == "2.7" ]; then
+    python -m SimpleHTTPServer 8888
+else
+    python -m http.server 8888
+fi
```
Comment (guillaumeeb) [inline ci/sge/run-master.sh]: Launching this http server is just in order to have a blocking process when running the docker container. I don't really like complexifying this line, even if I don't have a better solution yet (maybe just use some blocking commands, see https://stackoverflow.com/questions/2935183/bash-infinite-sleep-infinite-blocking).

File: ci/sge/Dockerfile-master (hunk 1 above)
Comment (kaelancotter) [reply ci/sge/Dockerfile-master]: ENV allows the run master/slave scripts access to TRAVIS_PYTHON_VERSION for the blocking script below.

File: ci/sge/run-master.sh (hunk 3 above)
Comment (kaelancotter) [reply ci/sge/run-master.sh]: I'll try out some variants based on above and can hopefully remove both above conditional and Docker ENV

File: ci/sge/docker-compose.yml (hunk 2 above)
Comment (kaelancotter) [reply ci/sge/docker-compose.yml]: Current behavior uses 3.6.6, but I don't know that its explicitly supported conda behavior for `python=""  `

**Comment by kaelancotter** [comment]
@guillaumeeb Is it alright to bump the docker-compose.yml version from 2 to 2.1?  Thats one option for default values in the compose stack.  [variable substitution](https://docs.docker.com/compose/compose-file/#variable-substitution) The other would be a `.env` file.  Replacement priority is listed here.  [env file](https://docs.docker.com/compose/environment-variables/#the-env-file).  If you or anyone else has a third suggestion I'm all ears.  Let me know which is preferable and I can hopefully get this PR wrapped up for ya.

**Comment by guillaumeeb** [comment]
>Is it alright to bump the docker-compose.yml version from 2 to 2.1?

Sure! Compose 2.1 is already two years old, moreover we use docker only for CI tests, so we don't need backward compatibility here.

**Comment by kaelancotter** [comment]
That should do it then.  Sorry about the crazy commit history.  I haven't done it before but I can try to rebase or something.

**Comment by guillaumeeb** [comment]
And no worries for the commits, we're using squash and merge functionnality of github.

**Pull request merged**